### DISCUSSION TITLE
How to remove last commits on github repo

### DISCUSSION THREAD

**Discussion by mstpha** [community/community#56789] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

How can i delete last n commits on my github repo ?

**Comment by waelbenmustapha** [comment]
To remove the last n commits from a GitHub repository and revert to a specific commit identified by its ID, you can use the following command:
git push -f origin <commitId>:master

replace commitId with the commit ID representing the last good state you want to return to.

**Comment by queenofcorgis** [comment]
@mstpha @waelbenmustapha GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answering and locking the post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
